docs/tutorial/beginner.rst

```diff
@@ -0,0 +1,572 @@
+
+SPRAS uses Snakemake, a workflow manager, together with Docker, containerized software, to read the configuration file and execute a SPRAS workflow. During a run, Snakemake will automatically fetch any missing Docker images as long as Docker is running. Snakemake considers a task from the configuration file complete once the expected output files are present in the output directory. As a result, rerunning the same configuration file may do nothing if those files already exist. If you want to continue running or rerun SPRAS with the same configuration file, remove the output directory (or its contents) or update the configuration file to trigger Snakemake to generate new results.
+
+For this part of the tutorial, we’ll use a pre-defined configuration file. 
+Download it here: :download:`Beginner Config File <../_static/config/beginner.yaml>`
+
+Save the file into the config/ folder of your SPRAS installation.
+After adding this file, SPRAS will use the configuration to set up and reference your directory structure, which will look like this:
+
+.. code-block:: text
+
+   spras/
+   ├── .snakemake/
+   │   └── log/
+   │       └── ... snakemake log files ...
+   ├── config/
+   │   └── basic.yaml
+   ├── inputs/
+   │   ├── phosphosite-irefindex13.0-uniprot.txt # pre-defined in SPRAS already
+   │   └── tps-egfr-prizes.txt # pre-defined in SPRAS already
+   ├── outputs/
+   │   └── basic/
+   │       └── ... output files ...
+
+
+Here’s an overview of the major sections looking at config/basic.yaml:
+
+Algorithms
+-----------
+
+
+.. code-block:: text
+    
+    algorithms:
+    - name: "pathlinker"
+        params:
+            include: true
+            run1:
+                k: 1
+            run2:
+                k: 10
+            run3:
+                k: [100, 400]
+
+
+When defining an algorithm in the configuration file, its name must match one of the supported wrapped algorithms within in SPRAS (I'll introduce the list of supported algorithms in the intermediate tutorial). Each algorithm includes an include flag, which you set to true to have Snakemake run it, or false to disable it. 
+
+The algorithm's parameters are grouped into one or more run blocks (e.g.  run1, run2, …). Within each run block, parameters are specified as key-value pairs. To define N runs, you can either create N separate run blocks, each with single parameter values, or use parameter lists within one (or multiple) run blocks, where the Cartesian product of those lists generates N parameter combinations. Each unique parameter combination is executed only once for the chosen algorithm, even if the same combination is defined multiple times. All parameter keys must be valid for that algorithm; unknown keys and missing required keys will cause SPRAS to fail.
```

Comment:
To define N runs, you can either create N separate run blocks, each with single parameter values, or use parameter lists within one (or multiple) run blocks, where the Cartesian product of those lists generates N parameter combinations.

This doesn't make sense when we only have K, maybe I need to update to a different algorithm or to add this cartisian product part to the next tutorial.